### ISSUE TITLE
Map OpenJDK options to OpenJ9 equivalents

### ISSUE THREAD

**Issue by DanHeidinga** [eclipse/openj9#2332]
`-XX:+HeapDumpOnOutOfMemoryError`

This is the default behaviour for OpenJ9.  The `-` form may be a shorter alias for disabling this behaviour.

`-XX:HeapDumpPath=<directory>`

We should consider aliasing this to `-Xdump:directory`, though that would change the directory for all java+system cores as well as snap+heap dumps.

**Comment by DanHeidinga** [comment]
See https://github.com/eclipse/openj9/pull/2323 for an example of how to do this kind of mapping

**Comment by pdbain-ibm** [comment]
Will we deprecate the OpenJ9 form of this option?

**Comment by DanHeidinga** [comment]
No, the intention of this work is to aid migration to OpenJ9.  Removing the existing OpenJ9 options would hurt current adopters.

**Comment by pdbain-ibm** [comment]
@DanHeidinga I have the changes for `-XX:HeapDumpPath=<directory>` tested and ready for review.  The changes for `-XX:+HeapDumpOnOutOfMemoryError` will take more thinking and work.  Would you like to review and merge the `-XX:HeapDumpPath=<directory>` changes now?

**Comment by DanHeidinga** [comment]
Two separate PRs makes sense to me - easier to review smaller chunks of code.

**Comment by pdbain-ibm** [comment]
I suggest that `-XX:+HeapDumpOnOutOfMemoryError` disable all dumps (heap, snap, java) on OOM.
@DanHeidinga @pshipton do you concur?

**Comment by pshipton** [comment]
@pdbain-ibm did you mean `-XX:-HeapDumpOnOutOfMemoryError` ?
+1 to have this disable all OOM dumps.

**Comment by pdbain-ibm** [comment]
Functional code merged.  Verifying the changes to the test code now.

**Comment by DanHeidinga** [comment]
FYI @vijaysun-omr - the two HeapDump related options we discussed have been merged.

**Comment by pdbain-ibm** [comment]
@vijaysun-omr and @DanHeidinga 
I believe this is complete.  May we close it?
Thank you
    -p

**Comment by vijaysun-omr** [comment]
Thanks for your work in getting these mappings done @pdbain-ibm 
I am unsure if this issue was meant to be a place to track Hotspot to OpenJ9 mappings on an ongoing basis or just for capturing the mappings that were mentioned above. 
If it's the latter then I have no issues closing it.

**Comment by DanHeidinga** [comment]
> meant to be a place to track Hotspot to OpenJ9 mappings on an ongoing basis

That's my intention for this issue.  A common place to track options that need to be mapped

**Comment by AlenBadel** [comment]
While running neo4j I noticed the following options used however they were not mapped to anything on openj9. 

-XX:+UseG1GC - Use the Garbage First (G1) Collector.
-XX:+PrintGCDetails - Prints all GC collections
-XX:+PrintGCDateStamps -  (Exists in Java 8 but removed in Java 9)
-XX:+PrintGCApplicationStoppedTime - Prints GC stop the world time, (safepoint time rather than gc)
-XX:+PrintPromotionFailure - Prints info about failures to move objects, pools of newer objects to pools of older
-XX:+PrintTenuringDistribution - Prints byte counts of age groups
-XX:+UseGCLogFileRotation - GC Log Rotation
-XX:GCLogFileSize - (Same as above)
-XX:NumberOfGCLogFiles (Same as above)
-XX:+AlwaysPreTouch - "JVM touches every single byte of the max heap size with a '0', resulting in the memory being allocated in the physical memory in addition to being reserved in the internal data structure" Rhel documentation
-XX:+OmitStackTraceInFastThrow - Omits some exceptions, removing some potentially unwanted info
-XX:+UnlockExperimentalVMOptions - is Oracle's option to enable unsupported options
-XX:+TrustFinalNonStaticFields - The ability to trust final fields, to constant fold final non-static fields.

**Comment by DanHeidinga** [comment]
More options to map in https://github.com/eclipse/openj9/issues/2834

**Comment by pshipton** [comment]
-XX:CompileThreshold is another one, I think it maps to -Xjit:count=

**Comment by pdbain-ibm** [comment]
Is https://github.com/eclipse/openj9/issues/2834 closed? Are there other outstanding OpenJDK options which we wish to map, beside `-XX:CompileThreshold`?

**Comment by pshipton** [comment]
-XX:NewSize, -XX:MaxNewSize, and -XX:ThreadStackSize
none of these "just worked" as mappings.

**Comment by gita-omr** [comment]
@zl-wang fyi

**Comment by DanHeidinga** [comment]
Another option mapping request: https://github.com/eclipse/openj9/issues/8184
